src/math/line.ts

```diff
@@ -1,20 +1,72 @@
-import Vector2 from './vector2.ts';
+import Vector2 from "./vector2.ts";
+import Rect from "./rect.ts"
+import {AABB, ShapeInterface} from "./shape.ts";
+import * as geometry from "./geometry-2d.ts";
 
-export default class Line {
-    start : Vector2 = Vector2.ZERO;
-    end : Vector2 = Vector2.ZERO;
+export default class Line implements ShapeInterface<Line>{
+    type: "Line" = "Line";
```

Comment:
```suggestion
    type: "Line" = "Line";
```

Voodoo, I don't know what to say